Merge master into bnotions branch, resolve conflicts and regression issues

rohan-dhamal-clarice opened the pull request:
I have merged master into <code>bnotions-merge</code> branch. Resolved conflicts. 
I also have created a build out of this branch and checked if social feature can be enabled/disabled with the flag. Looks all good.

However, there also seems to be minor regression. I will do the fixes on this branch itself before I merge this to master. 

Please review, @aleffert @nasthagiri .

rohan-dhamal-clarice commented:
@aleffert @nasthagiri Could you please confirm if bnotions has done any changes to the facebook SDK code? 
I am using Facebook gradle dependency rather than referencing Facebook SDK library in the project.

aleffert commented:
@rohan-dhamal-clarice I think that's fine.

rohan-dhamal-clarice commented:
@aleffert @nasthagiri I think we are good to merge this branch into master. Should I?

shahidtamboli commented on android/source/VideoLocker/build.gradle:
```diff
@@ -7,6 +7,7 @@ apply plugin: 'newrelic'
 
 repositories {
     mavenLocal()
+    maven { url "http://repo.maven.apache.org/maven2" }
```
Having mavenCentral() we had removed this. Do we still need it for this commit?

rohan-dhamal-clarice replied:
I will refine gradle scripts on a seperate branch.

shahidtamboli commented on android/source/VideoLocker/res/drawable/switch_track.xml:
```diff
@@ -1,5 +1,7 @@
+    <item android:drawable="@drawable/switch_disabled_on" android:state_enabled="false" android:state_checked="true"/>
     <item android:drawable="@drawable/switch_defaultbg" android:state_focused="true"/>
     <item android:drawable="@drawable/switch_cyanbg" android:state_checked="true"/>
     <item android:drawable="@drawable/switch_defaultbg" android:state_checked="false"/>
```
Default state needs to be handled.

rohan-dhamal-clarice replied:
Looks like the last line handles the default state. Please check the complete file.

shahidtamboli commented on android/source/VideoLocker/src/org/edx/mobile/module/db/impl/DbHelper.java:
```diff
@@ -53,8 +56,11 @@ public void onCreate(SQLiteDatabase db) {
+            String upgradeQuery = "ALTER TABLE "    + DbStructure.Table.DOWNLOADS
+                    + " ADD COLUMN " + DbStructure.Column.UNIT_URL + " TEXT "
+                    + " ADD COLUMN " + DbStructure.Column.URL_HIGH_QUALITY + " TEXT "
+                    + " ADD COLUMN " + DbStructure.Column.URL_LOW_QUALITY + " TEXT "
```
This might not work as the Db version is changed to 3 and the check for old version is 2.

rohan-dhamal-clarice replied:
That's a good catch. I correct this.

rohan-dhamal-clarice replied:
Looks like the query syntax is incorrect?

shahidtamboli commented on android/source/VideoLocker/src/org/edx/mobile/view/CourseListTabFragment.java:
```diff
@@ -0,0 +1,343 @@
+
+import java.util.List;
+
+public abstract class CourseListTabFragment extends Fragment implements NetworkObserver, MyCourseAdapter.CourseFriendsListener, LoaderManager.LoaderCallbacks<AsyncTaskResult<List<EnrolledCoursesResponse>>> {
```
Can we rename this to CourseListFragment?

rohan-dhamal-clarice replied:
May be, but skipping for now.

shahidtamboli commented:
@rohan-dhamal-clarice - A few points:
1. Remove -  android/source/VideoLocker/debug.keystore
2. Remove -  android/source/VideoLocker/libs/snakeyaml-android-1.8.jar
3. Do we require this  -  android/source/VideoLocker/libs/android-support-v4.jar
4. We will require to refactor some images to 9 patch later on when enabling the features.

I have also added a few comments.

edx-webhook commented:
Thanks for the pull request, @rohan-dhamal-clarice! I've created [OSPR-394](https://openedx.atlassian.net/browse/OSPR-394) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams. 

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ('this must be merged by XX date', and why that is)
- partner information ('this is a course on edx.org')
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the Github pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

We can't start reviewing your pull request until you've added yourself to the [AUTHORS](https://github.com/edx/edx-app-android/blob/master/AUTHORS) file. Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst) file for more information.

sarina commented:
I don't have access to this repo, unfortunately, but please feel free to delete the bot's post above and my post here as well.

aleffert commented:
@rohan-dhamal-clarice If you think it's ready. I think you should go ahead and merge. Just to confirm, we're hoping for the following things before it's merged:

1. When the social features flag is disabled (which is the default), the only Facebook feature is log in.
2. It's been through a QA pass and is reasonably stable.

If those conditions are met. Go ahead.

rohan-dhamal-clarice commented:
@shahidtamboli 
1. Removed - android/source/VideoLocker/debug.keystore
2. Removed as not required - android/source/VideoLocker/libs/snakeyaml-android-1.8.jar
3. Removed (using gradle dependency) - android/source/VideoLocker/libs/android-support-v4.jar

For images, we will check if we can replace some of them with 9 patch.
